src/main/java/com/PBA/authservice/validator/UserValidatorImpl.java

```diff
@@ -0,0 +1,105 @@
+package com.pba.authservice.validator;
+
+import com.pba.authservice.controller.request.UserCreateRequest;
+import com.pba.authservice.exceptions.AuthValidationException;
+import com.pba.authservice.persistance.model.PendingUser;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Component
+public class UserValidatorImpl implements UserValidator {
+    @Override
+    public void validateUserCreateRequest(UserCreateRequest userCreateRequest) {
+        Map<String, String> errorMap = new LinkedHashMap<>();
```

Comment:
I already discussed with you to use `spring-starter-validation` https://www.baeldung.com/spring-boot-bean-validation with specific annotations like `@NotNull`, `@NotBlank`, `@Size` etc. You can use those to check for constraints and return a error message.
and to inject the error messages from a file from resources, like error_messages.properties

Remove the entire code, you don't need it.